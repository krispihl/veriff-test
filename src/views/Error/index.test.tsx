import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ErrorView } from './index';

describe('Error', () => {
    const { location } = window;

    beforeAll(() => {
        // @ts-ignore
        delete window.location;
        // @ts-ignore
        window.location = { reload: jest.fn() };
    });

    afterAll(() => {
        window.location = location;
    });

    it('renders Error view correctly', () => {
        const { container } = render(<ErrorView />);

        expect(container).toMatchSnapshot();
        expect(screen.getByText('Whoops! Something went wrong...', { exact: true})).toBeInTheDocument();
        expect(screen.getByRole('button')).toBeInTheDocument();
        expect(screen.getByRole('link')).toBeInTheDocument();
    });

    it('window reload is called when user clicks try again button', () => {
        render(<ErrorView />);

        const button = screen.getByRole('button');

        userEvent.click(button);

        expect(window.location.reload).toHaveBeenCalled();
    });
})