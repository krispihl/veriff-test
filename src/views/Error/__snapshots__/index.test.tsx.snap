// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders Error view correctly 1`] = `
<div>
  <div
    class="container"
  >
    <h2>
      Whoops! Something went wrong...
    </h2>
    <div>
      Please 
      <button>
        try again
      </button>
       or contact our 
      <a
        href="https://support.veriff.com/en"
      >
        support.
      </a>
    </div>
  </div>
</div>
`;
